"""
Protein structure descriptions as per PubMed.
Each variable contains Markdown/HTML-formatted text for display in the Dash app.
"""

desc_1CRN = """
## 1CRN
#### PDB ID: [1CRN](https://www.rcsb.org/structure/1CRN)
#### PubMed ID: [16593516](https://pubmed.ncbi.nlm.nih.gov/16593516/)
---
<h4>PubMed Abstract</h4>
<p>
The water structure has been analyzed for a model of the protein crambin
refined against 0.945-A x-ray diffraction data. Crystals contain 32% solvent
by volume, and 77% of the solvent molecules have been located-i.e., 2 ethanol
molecules and 64 water molecules with 10-14 alternate positions. Many water
oxygen atoms found form chains between polar groups on the surface of the
protein. However, a cluster of pentagonal arrays made up of 16 water molecules
sits at a hydrophobic, intermolecular cleft and forms a cap around the methyl
group of leucine-18. Several waters in the cluster are hydrogen-bonded directly
to the protein. Additional closed circular arrays, which include both protein
atoms and other water oxygen atoms, form next to the central cluster. This water
array stretches in the b lattice direction between groups of three ionic side
chains.
</p>
"""

desc_1AAZ = """
## 1AAZ
### PDB ID : [1AAZ](https://www.rcsb.org/structure/1AAZ)
### PubMed ID : [1453466](https://pubmed.ncbi.nlm.nih.gov/1453466/)
---
<h4>PubMed Abstract</h4>
<p>
The structure of wild-type bacteriophage T4 glutaredoxin (earlier called
thioredoxin) in its oxidized form has been refined in a monoclinic crystal form
at 2.0 A resolution to a crystallographic R-factor of 0.209. A mutant T4
glutaredoxin gives orthorhombic crystals of better quality. The structure of
this mutant has been solved by molecular replacement methods and refined at
1.45 A to an R-value of 0.175. In this mutant glutaredoxin, the active site
residues Val15 and Tyr16 have been substituted by Gly and Pro, respectively, to
mimic that of Escherichia coli thioredoxin. The main-chain conformation of the
wild-type protein is similar in the two independently determined molecules in
the asymmetric unit of the monoclinic crystals. On the other hand, side-chain
conformations differ considerably between the two molecules due to heterologous
packing interactions in the crystals. The structure of the mutant protein is
very similar to the wild-type protein, except at mutated positions and at parts
involved in crystal contacts. The active site disulfide bridge between Cys14 and
Cys17 is located at the first turn of helix alpha 1. The torsion angles of these
residues are similar to those of Escherichia coli thioredoxin. The torsion angle
around the S-S bond is smaller than that normally observed for disulfides: 58
degrees, 67 degrees and 67 degrees for wild-type glutaredoxin molecule A and B
and mutant glutaredoxin, respectively. Each sulfur atom of the disulfide
cysteines in T4 glutaredoxin forms a hydrogen bond to one main-chain nitrogen
atom. The active site is shielded from solvent on one side by the beta-carbon
atoms of the cysteine residues plus side-chains of residues 7, 9, 21 and 33.
From the opposite side, there is a cleft where the sulfur atom of Cys14 is
accessible and can be attacked by a nucleophilic thiolate ion in the initial
step of the reduction reaction.
</p>
"""

desc_1AG6 = """
## 1AG6
### PDB ID : [1AG6](https://www.rcsb.org/structure/1AG6)
### PubMed ID : [9792096](https://pubmed.ncbi.nlm.nih.gov/9792096/)
---
<h4>PubMed Abstract</h4>
<p>
The crystal structure of plastocyanin from spinach has been determined using
molecular replacement, with the structure of plastocyanin from poplar as a
search model. Successful crystallization was facilitated by site-directed
mutagenesis in which residue Gly8 was substituted with Asp. The region around
residue 8 was believed to be too mobile for the wild-type protein to form
crystals despite extensive screening. The current structure represents the
oxidized plastocyanin, copper (II), at low pH (approximately 4.4). In contrast
to the similarity in the core region as compared to its poplar counterpart, the
structure shows some significant differences in loop regions. The most notable
is the large shift of the 59-61 loop where the largest shift is 3.0 A for the
C(alpha) atom of Glu59. This results in different patterns of electrostatic
potential around the acidic patches for the two proteins.
</p>
"""

desc_1A7I = """
## 1A7I
### PDB ID : [1A7I](https://www.rcsb.org/structure/1A7I)
### PubMed ID : [9585524](https://pubmed.ncbi.nlm.nih.gov/9585524/)
---
<h4>PubMed Abstract</h4>
<p>
Members of the cysteine and glycine-rich protein (CRP) family (CRP1, CRP2, and
CRP3) contain two zinc-binding LIM domains, LIM1 and LIM2, and are implicated in
diverse cellular processes linked to differentiation, growth control and
pathogenesis. The solution structure of an 81-amino acid recombinant peptide
encompassing the amino-terminal LIM1 domain of quail CRP2 has been determined by
2D and 3D homo- and heteronuclear NMR spectroscopy. The LIM1 domain consists of
two zinc binding sites of the CCHC and the CCCC type, respectively, which both
contain two orthogonally arranged antiparallel beta-sheets and which are packed
together by a hydrophobic core composed of residues from the zinc finger loop
regions. The CCCC zinc finger is followed by a short alpha-helical stretch. The
structural analysis revealed that the global fold of LIM1 closely resembles the
recently determined solution structures of the carboxyl-terminal LIM2 domains of
quail CRP2 and chicken CRP1, and that LIM1 and LIM2 are independently folded
structural and presumably functional domains of CRP proteins. To explore the
dynamical properties of CRP proteins, we have used 15N relaxation values (T1,
T2, and nuclear Overhauser effect (NOE) to describe the dynamical behavior of a
LIM domain. A model-free analysis revealed local variations in mobility along
the backbone of the quail CRP2 LIM1 motif. Slow motions are evident in turn
regions located between the various antiparallel beta-sheets or between their
strands. By use of an extended motional model, fast backbone motions were
detected for backbone amide NH groups of hydrophobic residues located in the
core region of the LIM1 domain. These findings point to a flexible hydrophobic
core in the LIM1 domain allowing residual relative mobility of the two zinc
fingers, which might be important to optimize the LIM1 interface for
interaction with its physiological target molecule(s) and to compensate
enthalpically for the entropy loss upon binding.
</p>
"""

desc_1AL1 = """
## 1AL1
### PDB ID : [1AL1](https://www.rcsb.org/structure/1AL1)
### PubMed ID : [2382133](https://pubmed.ncbi.nlm.nih.gov/2382133/)
---
<h4>PubMed Abstract</h4>
<p>
X-ray diffraction shows the structure of a synthetic protein model, formed from
noncovalent self-association of a 12-residue peptide and of sulfate ions at low
pH. This peptide is a fragment of a 16-residue polypeptide that was designed to
form an amphiphilic alpha helix with a ridge of Leu residues along one helical
face. By interdigitation of the leucines of four such helices, the design called
for self-association into a four-alpha-helical bundle. The crystal structure
(2.7 angstrom resolution; R factor = 0.215) reveals a structure more complex
than the design, with both a tetramer and a hexamer. The alpha-helical tetramer
with leucine interior has more oblique crossing angles than most four-alpha-
helical bundles; the hexamer has a globular hydrophobic core of 12 leucine
residues and three associated sulfate ions. Computational analysis suggests that
the hexameric association is tighter than the tetrameric one. The consistency of
the structure with the design is discussed, as well as the divergence.
</p>
"""

desc_1AOJ = """
## 1AOJ
### PDB ID : [1AOJ](https://www.rcsb.org/structure/1AOJ)
### PubMed ID : [9303002](https://pubmed.ncbi.nlm.nih.gov/9303002/)
---
<h4>PubMed Abstract</h4>
<p>
SH3 domains are structurally well-characterized as monomeric modular units of
protein structure that mediate protein-protein recognition in numerous signal
transduction proteins. The X-ray crystallographic structure of the Eps8 SH3
domain reveals a novel variation of the canonical SH3 fold: the SH3 domain from
Eps8 is a dimer formed by strand interchange. In addition, co-immunoprecipitation
experiments show that intact Eps8 is multimeric in vivo. Hence, the SH3 domain
of Eps8 may represent a dimerization motif.
</p>
"""

desc_1A32 = """
## 1A32
### PDB ID : [1A32](https://www.rcsb.org/structure/1A32)
### PubMed ID : [9562554](https://pubmed.ncbi.nlm.nih.gov/9562554/)
---
<h4>PubMed Abstract</h4>
<p>
Ribosomal protein S15 is a primary RNA-binding protein that binds to the central
domain of 16S rRNA. S15 also regulates its own synthesis by binding to its own
mRNA. The binding sites for S15 on both mRNA and rRNA have been narrowed down to
less than a hundred nucleotides each, making the protein an attractive candidate
for the study of protein-RNA interactions. The crystal structure of S15 from
Bacillus stearothermophilus has been solved to 2.1 A resolution. The structure
consists of four alpha helices. Three of these helices form the core of the
protein, while the N-terminal helix protrudes out from the body of the molecule
to make contacts with a neighboring molecule in the crystal lattice. S15
contains a large conserved patch of basic residues which could provide a site
for binding 16S rRNA. The conformation of the N-terminal alpha helix is quite
different from that reported in a recent NMR structure of S15 from Thermus
thermophilus. The intermolecular contacts that this alpha helix makes with a
neighboring molecule in the crystal, however, closely resemble the intramolecular
contacts that occur in the NMR structure. This conformational variability of the
N-terminal helix has implications for the range of possible S15-RNA interactions.
A large, conserved basic patch at one end of S15 and a cluster of conserved but
exposed aromatic residues at the other end provide two possible RNA-binding sites
on S15.
</p>
"""

desc_about = """
### *Computational Geometry of three dimensional protein structures.*
---

Convex Hulls and Delaunay Tessellations are two of the most commonly used
algorithms in the field of Computational Geometry. Several researchers have
applied these algorithms for the study of protein three dimensional structure
and function. This application aims to serve as a simple visual aid to
interactively study the protein structure using the molecule3d viewer, a
computational geometry chart, ramachandran chart and the associated features
resulting from the application of the two algorithms as a table.

The layout of the application is pretty simple. The user can select one of the
seven available protein data bank structures for visualization (using the Plotly
Molecule3dViewer from the Dash Bio library), interact with the topological
constructs using the computational geometry chart and view the nature of the
peptide bond in the ramachandran chart. The Computational Geometry and
Ramachandran Chart are generated using Plotly.

The application of Convex Hulls and Delaunay Tessellation to the three
dimensional point cloud results in a series of features:
 - amino_acid_residues: residues participating in the Delaunay Tetrahedra
 - simplex_indices: residue indexes of the amino_acid_residues
 - simplex_type: The type of Delaunay Tetrahedra formed, which depends on the
   number of contiguous residues. Type 0 for no contiguous residues to Type 4
   for all 4 residues contiguous.
 - L1, L2, L3, L4, L5, L6: Edge lengths of the Delaunay Tetrahedra
 - SumL: Sum of Edge Lengths
 - AvgL: Average Edge Lengths
 - TVol: Delaunay Tetrahedra Volume
 - T1_area, T2_area, T3_area, T4_area: Surface area of the triangles of the
   Delaunay Tetrahedra
 - Tsurface_area: Surface area of the tetrahedra.
 - O1, O2, O3, O4: Solid angles formed at the vertices of the Delaunay Tetrahedra.
 - The Convex Hull area and volume are added as annotations to the chart.

In the interactive mode, the users can only interact with the C-alpha Delaunay
Tessellation dataset, but the datasets for C-alpha and all atom Delaunay
Tessellation & C-alpha and all atom Convex Hulls are available for download from
the dashboard (as well as the PDB file).

For the sake of simplicity, only single chain (chain A) of the proteins is
included in the application and the residue length of the protein chains is
limited to about 175. But protein chains with diverse structure and function
have been included:
 - 1A71: an amino terminal LIM domain from quail - 81 residues
 - 1A32: ribosomal protein from Bacillus Stearothermophilus - 88 residues
 - 1AAZ: bacteropphage T4 glutaredoxin - 174 residues
 - 1AG6: plastocyanin from spinach - 99 residues
 - 1AL1: a synthetic protein model - 81 residues
 - 1AOJ: SH3 domain - 120 residues
 - 1CRN: a hydrophobic plant protein - 46 residues

**Note:**
 - The application is best viewed on a laptop or a desktop.
 - Upon selecting a residue in the molecular3dviewer, the corresponding residue
   will be highlighted in the ramachandran chart and all the residues
   participating in the C-alpha Delaunay Tessellation will also be highlighted.
 - The PDB files were altered to set the center of mass to zero and residue
   numbering starting from zero.
 - Only single chain (chain A) of the protein structures was used.
 - For the Ramachandran Chart, only the residues containing the Φ and Ψ angles
   are considered. So the first and last residue will not be displayed. But
   these residues will be highlighted in the Computational Geometry chart and
   will be displayed in the filtered data table.
 - The "Reset View" will clear all the user selections, but this may take a few
   seconds.
 - Use the Plotly chart controls to reset individual charts.
 - The code is available on GitHub.

---
##### Resources used for creating this application : [Dash Plotly](https://dash.plotly.com/), [Dash Bio](https://dash.plotly.com/dash-bio), [Dash Mantine Components](https://www.dash-mantine-components.com/), [Dash Bootstrap](https://www.dash-bootstrap-components.com/), [Biopandas](https://biopandas.github.io/biopandas/), [Pandas](https://pandas.pydata.org/), [Biopython](https://biopython.org/), [Python](https://www.python.org/), [ChatGPT](https://chatgpt.com/)
"""